sklearn/ensemble/tests/test_stacking.py

```diff
@@ -491,3 +498,76 @@ def test_stacking_cv_influence(stacker, X, y):
     with pytest.raises(AssertionError, match='Not equal'):
         assert_allclose(stacker_cv_3.final_estimator_.coef_,
                         stacker_cv_5.final_estimator_.coef_)
+
+
+class ProdClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self, idx=None):
+        self.idx = idx
+        self.n_features_in_ = 0
+
+    def fit(self, X, y=None):
+        return self
+
+    def predict(self, X):
+        if not self.idx:
+            return np.prod(X, axis=1)
+        else:
+            return np.prod(X[:, self.idx], axis=1)
+
+
+class SumRegressor(BaseEstimator, RegressorMixin):
+    def __init__(self, idx=None):
+        self.idx = idx
+        self.n_features_in_ = 0
+
+    def fit(self, X, y=None):
+        return self
+
+    def predict(self, X):
+        if not self.idx:
+            return np.sum(X, axis=1)
+        else:
+            return np.sum(X[:, self.idx], axis=1)
+
+
+@pytest.mark.parametrize(
+    "stacker, X, y",
+    [(StackingClassifier(
+        estimators=[('d0', ProdClassifier([0])),
+                    ('d1', ProdClassifier([1]))],
+        cv="prefit",
+        final_estimator=ProdClassifier([0, 1])),
+      X_dummy_classification, y_dummy_classification),
+     (StackingRegressor(
+         estimators=[('d0', SumRegressor([0])),
+                     ('d1', SumRegressor([1]))],
+         cv="prefit",
+         final_estimator=SumRegressor([0, 1])),
+      X_dummy_regression, y_dummy_regression)],
+    ids=['StackingClassifier', 'StackingRegressor']
+)
+def test_stacking_prefit(stacker, X, y):
```

Comment:
Actually I've realised that this does not test that the "prefitted" models are used: fit could still be being called; indeed `cross_val_predict` could still be used and this would pass, AFACT.

Reply:
Thanks for the catch! I modified my dummy classifier and regressor a little bit so that they would raise an `AttributeError` if used as base estimator in the stacker by setting `is_base_estimator=True`, so that we are sure fit is not called upon them.